src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -45,56 +57,43 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
-                ProjectPackagesPrintUtility.PrintSources(listPackageArgs.PackageSources);
-                Console.WriteLine();
-            }
-
-            foreach (var projectPath in projectsPaths)
-            {
-                //Open project to evaluate properties for the assets
-                //file and the name of the project
-                var project = MSBuildAPIUtility.GetProject(projectPath);
+                // Project specific data stored in below variable
+                ListPackageProjectModel projectModel = listPackageReportModel.CreateProjectReportData(projectPath: projectPath);
+                Project project = projectModel.Project;
 
                 if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
                 {
-                    Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
-                        Strings.Error_NotPRProject,
-                        projectPath));
-                    Console.WriteLine();
+                    projectModel.AddProjectProblem(error: string.Format(CultureInfo.CurrentCulture,
+                        Strings.Error_NotPRProject, projectPath),
+                        problemType: ProblemType.Error);
                     continue;
                 }
 
-                var projectName = project.GetPropertyValue(ProjectName);
-
                 var assetsPath = project.GetPropertyValue(ProjectAssetsFile);
 
                 // If the file was not found, print an error message and continue to next project
                 if (!File.Exists(assetsPath))
                 {
-                    Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
-                        Strings.Error_AssetsFileNotFound,
-                        projectPath));
-                    Console.WriteLine();
+                    projectModel.AddProjectProblem(error: string.Format(CultureInfo.CurrentCulture,
+                        Strings.Error_AssetsFileNotFound, projectPath),
+                        problemType: ProblemType.Error);
                 }
                 else
                 {
                     var lockFileFormat = new LockFileFormat();
-                    var assetsFile = lockFileFormat.Read(assetsPath);
+                    LockFile assetsFile = lockFileFormat.Read(assetsPath);
 
                     // Assets file validation
                     if (assetsFile.PackageSpec != null &&
                         assetsFile.Targets != null &&
                         assetsFile.Targets.Count != 0)
                     {
                         // Get all the packages that are referenced in a project
-                        var packages = msBuild.GetResolvedVersions(project.FullPath, listPackageArgs.Frameworks, assetsFile, listPackageArgs.IncludeTransitive, includeProjects: listPackageArgs.ReportType == ReportType.Default);
+                        IEnumerable<FrameworkPackages> packages = projectModel.GetAssetFilePackages(assetsFile);
```

Comment:
The method should probably be in this class.

Reply:
Ok. Moved back to this file.